scheduler/src/cook/mesos/api.clj

```diff
@@ -935,6 +946,23 @@
       (log/debug e "Unable to retrieve directory path for" task-id "on agent" agent-hostname)
       nil)))
 
+(defn compute-cluster-entity->map
+  [entity]
+  (cond-> {:compute-cluster-name (:compute-cluster/cluster-name entity)}
+    (= :compute-cluster.type/mesos (:compute-cluster/type entity))
+          (assoc :mesos {:framework-id (:compute-cluster/mesos-framework-id entity)})))
+
+(defn fetch-compute-cluster-map
+  "Converts a compute cluster entity as a map representing the fields. For legacy instances
+  that predate cook-cluster, inline the default cluster, pulled from config."
+  [db entity]
+  (if entity
+    (compute-cluster-entity->map entity)
+    (->> (cc/cluster-name-hack)  ; Get the default cluster.
+         cc/cluster-name->db-id
```

Comment:
Isn't there an atom with the compute cluster entities cached? should we just use that?

Reply:
I wrapped the atom with that function; I want that atom to be an internal implementation detail.